src/serialization/json_object.cpp

```diff
@@ -265,8 +268,18 @@ void fromJsonValue(const rapidjson::Value& val, cryptonote::transaction& tx)
   GET_FROM_JSON_OBJECT(val, tx.vin, inputs);
   GET_FROM_JSON_OBJECT(val, tx.vout, outputs);
   GET_FROM_JSON_OBJECT(val, tx.extra, extra);
-  GET_FROM_JSON_OBJECT(val, tx.signatures, signatures);
   GET_FROM_JSON_OBJECT(val, tx.rct_signatures, ringct);
+
+  const auto& sigs = val.FindMember("signatures");
+  if (sigs != val.MemberEnd())
+  {
+    fromJsonValue(sigs->value, tx.signatures);
+  }
+
+  const auto& rsig = tx.rct_signatures;
+  const bool is_coinbase = !tx.vin.empty() && boost::get<txin_gen>(std::addressof(tx.vin[0]));
```

Comment:
FWIW there's a in_coinbase in cryptonote:: and it checks for vin.size() == 1, not just !empty. I'm not sure it's really that important though.

Reply:
Probably fine, since invalid txes will have got rejected before that.

Reply:
Now using `cryptonote::is_coinbase`.